skimage/morphology/rolling_ball.py

```diff
@@ -0,0 +1,197 @@
+import numpy as np
+from numpy.lib.stride_tricks import as_strided
+
+from skimage.util import invert, view_as_windows
+from ._rolling_ball_cy import apply_kernel, apply_kernel_nan
+
+
+def rolling_ellipsoid(image, kernel_shape=100, intensity_vertex=100,
+                      has_nan=False, num_threads=None):
+    """
+    Estimate background intensity using a rolling ellipsoid.
+
+    The rolling ellipsoid algorithm estimates background intensity for a
+    grayscale image in case of uneven exposure. It is a generalization of the
+    frequently used rolling ball algorithm [1]_.
+
+    Parameters
+    ----------
+    image : ndarray
+        The image to be filtered.
+    kernel_shape: ndarray or scalar, optional
+        The length of the non-intensity vertices of the ellipsoid. If
+        ``kernel_shape`` is a ndarray, it must have the same dimensions as
+        ``image``. If ``kernel_shape`` is a scalar it will be extended to
+        have the same dimension via
+        ``kernel_shape = kernel_shape * np.ones_like(image.shape)``.
+        All elements must be greater than 0.
+    intensity_vertex : scalar, optional
+        The length of the intensity vertex of the ellipsoid. Must be greater
```

Comment:
Why not radius for this? "length of the intensity vertex" is super confusing to me.

Reply:
Because it is the "diameter" ... but only in the special case if the ellipsoid is a ball. In the general case, the terms radius and diameter are non-sensical.

Reply:
I renamed vertex to axis. This seems more consistent